### ISSUE TITLE
Enable the specification of the Help flag's shorthand character and usage value

### ISSUE THREAD

**Issue by akutz** [spf13/cobra#129]
I would like to be able to specify the value used for the help flag's shorthand character so that when there's a global "host" flag, it can use the "h" for its shorthand and the help flag can use "?". Additionally, the ability to format the help flag's usage value would also be nice for applications with different casing standards for help output.

It appears that the help flag is automatically generated as part of the core cobra package (when a flag set is nil), so I am submitting a patch that should make it easy for this routine to take into account user preference.

**Comment by eparis** [comment]
A work around would be to define the `help` flag yourself with either a different shorthand or no shorthand. Cobra's help will only get added if you didn't do it...

**Comment by akutz** [comment]
Well crap on a stick, how did I miss that? I saw you can do that for the command, but not the flag. I tried visiting all flags for all commands, setting each help flag's shorthand to what I wanted when I matched the name, but still the panic.

FWIW, here's the commit with which I was going to open a pull request. It doesn't seem necessary now. I'll test your suggestion and if it works, close this issue.

**Comment by eparis** [comment]
(I could be wrong, wouldn't be the first time)

**Comment by akutz** [comment]
Are you sure you're not thinking of the command? The source for how the help flag is created doesn't seem to allow for the previous existence of said flag:

```
// Get the complete FlagSet that applies to this command (local and persistent declared here and by all parents)
func (c *Command) Flags() *flag.FlagSet {
	if c.flags == nil {
		c.flags = flag.NewFlagSet(c.Name(), flag.ContinueOnError)
		if c.flagErrorBuf == nil {
			c.flagErrorBuf = new(bytes.Buffer)
		}
		c.flags.SetOutput(c.flagErrorBuf)
		c.PersistentFlags().BoolVarP(&c.helpFlagVal, "help", "h", false, "help for "+c.Name())
	}
	return c.flags
}
```

You cannot access Flags without forcing the creation of the help flag from what I'm able to tell.

**Comment by akutz** [comment]
It doesn't look like you can do that...

```
[0]akutz@poppy:rexray$ rexray
panic: rexray flag redefined: help

goroutine 1 [running]:
github.com/spf13/pflag.(*FlagSet).AddFlag(0xc8200f0580, 0xc82011c240)
	/home/akutz/go/src/github.com/spf13/pflag/flag.go:522 +0x3f6
github.com/spf13/pflag.(*FlagSet).VarPF(0xc8200f0580, 0x7fd2545e8230, 0xd13178, 0xa06d30, 0x4, 0xa06c00, 0x1, 0xc8200d3db0, 0xf, 0xf)
	/home/akutz/go/src/github.com/spf13/pflag/flag.go:504 +0x11e
github.com/spf13/pflag.(*FlagSet).BoolVarP(0xc8200f0580, 0xd13178, 0xa06d30, 0x4, 0xa06c00, 0x1, 0x0, 0xc8200d3db0, 0xf)
	/home/akutz/go/src/github.com/spf13/pflag/bool.go:58 +0xb2
github.com/emccode/rexray/vendor/github.com/spf13/cobra.(*Command).Flags(0xd13000, 0x5)
	/home/akutz/go/src/github.com/emccode/rexray/vendor/github.com/spf13/cobra/command.go:864 +0x355
github.com/emccode/rexray/rexray/commands.initGlobalFlags()
	/home/akutz/go/src/github.com/emccode/rexray/rexray/commands/rexraycli.go:715 +0x2f
github.com/emccode/rexray/rexray/commands.initFlags()
	/home/akutz/go/src/github.com/emccode/rexray/rexray/commands/rexraycli.go:706 +0x8a
github.com/emccode/rexray/rexray/commands.init.1()
	/home/akutz/go/src/github.com/emccode/rexray/rexray/commands/rexraycli.go:687 +0x1e
github.com/emccode/rexray/rexray/commands.init()
	/home/akutz/go/src/github.com/emccode/rexray/rexray/commands/rexraycli.go:1291 +0x449
main.init()
	/home/akutz/go/src/github.com/emccode/rexray/rexray/rexraycli.go:7 +0x3b

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/opt/go/1.5/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/opt/go/1.5/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/opt/go/1.5/src/os/signal/signal_unix.go:28 +0x37
```

**Comment by eparis** [comment]
#130 and #133 are possibilities...

**Comment by eparis** [comment]
"fixed" with #133

**Comment by akutz** [comment]
Hi @eparis,

Your change in #133 was reverted at some point making my simpler fix once again necessary...
